src/cli/api/local_api.py

```diff
@@ -382,20 +526,266 @@ def get_results(self, path: Path, all_results: bool) -> List[ResultType]:
+                    self.validate_experiment_nodes(experiment_file_path, experiment_data, error_dict)
+                    # Validate experiment channels
+                    self.validate_experiment_channels(experiment_file_path, experiment_data, error_dict)
+                else:
+                    error_dict['warning'].append(f"In file {experiment_file_path}: network '{experiment_network_slug}' "
+                                                 f"does not exist")
+
+                self.validate_experiment_application(path, experiment_data, error_dict)
+
+        # Validate application.json, network.json and result.json
+        if os.path.exists(experiment_input_path):
+            app_schema_path = Path(os.path.join(BASE_DIR), 'schema', 'applications', '')
+            files_list = ["application.json", "network.json", "result.json"]
+
+            for file in files_list:
+                if os.path.isfile(experiment_input_path / file):
+                    json_valid, message = validate_json_file(experiment_input_path / file)
+                    if json_valid:
+                        schema_valid, ve = validate_json_schema(experiment_input_path / file, app_schema_path / file)
+                        if not schema_valid:
+                            error_dict['error'].append(ve)
+                    else:
+                        error_dict['error'].append(message)
+
+    def validate_experiment_nodes(self, experiment_file_path: Path, experiment_data: Dict[str, Any], error_dict:
+                                  ErrorDictType) -> None:
+        """
+        Validate if the amount of nodes (defined in the experiment.json file) are valid for 'network_slug' and if
+        all the nodes exist and belong to this network.
+
+        Args:
+            experiment_file_path: The location of the experiment.json file
+            experiment_data: contents of the experiment.json file
+            error_dict: Dictionary containing error and warning messages of the validations that failed
+        """
+        experiment_network_slug = experiment_data['asset']['network']['slug']
+
+        # Check if the amount of nodes are valid for this network
+        experiment_nodes = experiment_data['asset']['network']['nodes']
+        network_nodes = self._get_network_nodes(experiment_network_slug)
+        if len(experiment_nodes) > len(network_nodes):
+            error_dict['warning'].append(f"In file {experiment_file_path}: too many nodes used in network "
+                                         f"'{experiment_network_slug}'. Maximum amount of nodes that can be used: "
+                                         f"{len(network_nodes)}")
+
+        # Check if all nodes exist and belong to this network
+        invalid_nodes = []
+        for node in experiment_nodes:
+            if node['slug'] not in network_nodes:
+                invalid_nodes.append(node['slug'])
+        if invalid_nodes:
+            if len(invalid_nodes) > 1:
```

Comment:
Instead of this check for plural form etc. I opt for adding a line to the error_dict for each wrong node.
You can already do that in line 709. You don't need the invalid_nodes list. Also printing a list you get the [], which we would not have when printing each message separate.

Reply:
The same still counts  for missing_files in validate_experiment_structure